frontends/web/src/routes/router.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { FunctionComponent } from "react";
+import { Route, Routes, useParams } from "react-router";
+import { IAccount } from '../api/account';
+import { TDevices } from '../api/devices';
+import { AddAccount } from './account/add/add';
+import { Moonpay } from './buy/moonpay';
+import { BuyInfo } from './buy/info';
+import { Info } from './account/info/info';
+import { Receive } from './account/receive/receive';
+import { Send } from './account/send/send';
+import { AccountsSummary } from './account/summary/accountssummary';
+import { DeviceSwitch } from './device/deviceswitch';
+import ManageBackups from './device/manage-backups/manage-backups';
+import { ManageAccounts } from './settings/manage-accounts';
+import { Exchanges } from './exchanges/exchanges';
+import ElectrumSettings from './settings/electrum';
+import { Settings } from './settings/settings';
+import { Passphrase } from './device/bitbox02/passphrase';
+import { Account } from "./account/account";
+
+interface Props {
+    devices: TDevices;
+    deviceIDs: string[];
+    accounts: IAccount[];
+    activeAccounts: IAccount[];
+    devicesKey: ((input: string) => string)
+}
+
+const InjectParams: FunctionComponent = ({ children }) => {
+    const params = useParams();
+    return React.cloneElement(children as React.ReactElement, params)
+}
+
+export const AppRouter: FunctionComponent<Props> = ({ devices, deviceIDs, devicesKey, accounts, activeAccounts }) => {
+    const Homepage = <DeviceSwitch
+        default
+        key={devicesKey('device-switch-default')}
+        deviceID={null}
+        devices={devices}
+    />
+
+    const Device = <InjectParams>
+        <DeviceSwitch
+            key={devicesKey('device-switch')}
+            deviceID={null}
+            devices={devices} />
+    </InjectParams>
+
+    const Acc = <InjectParams>
+        <Account
+            code={'' /* dummy to satisfy TS */}
```

Comment:
Without changing the components, yes, because whe are expecting the params to be defined in the Props of this components. This would make defining params like {code?: string}, but we know that the params are always injected, so it would be bad.